### PR TITLE
fix(vuesim) : Combinational Analysis tool reads the input from table

### LINKED ISSUE #1307: 🐞 Bug: [vuesim] The combinational analysis fails to generate circuit
### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Describe the bug

The combinational analysis tool doesnt generate the circuit. 
further the  user is unable to enter the values in the truth table for the output gate  

### Steps to Reproduce

1. enable vuesim in flipper
2. go to tools > combinational analysis
3. enter the input gate (  a, b ) and output gate as (y)
4. click next
5.  fails to change the value for output gate

### Expected Behavior

same as legacy simulator and be able to generate the circuit

### Screenshots

_No response_

### Is the faced issue/bug related to the Vue simulator?

Yes

### Vue simulator version

v0

### Used Vue simulator with or without backend?

Without Backend

### Is the bug present only on the dev server, the build, or both?

Both

### Device Information

```markdown

```

### Additional Context

_No response_

### Are you working on this issue?

Yes

### PR DESCRIPTION
**Pull request by ShinichiShi**
Fixes #1307

 

#### Describe the changes you have made in this PR -
The Combinational Anaysis dialog was completely broken in both Vue due to four bugs introduced when the feature was ported from the old jQuery-based simulator:

- `GenerateCircuit()` was called with zero arguments in `CombinationalAnalysis` file, even though it requires 4–7 parameters. Therefore failing "Generate Circuit".
- `generateBooleanTableData()` scraped the DOM via jQuery ($('.${outputName}')) looking for <td> cells with a class/id scheme that no longer exists in the Vue-rendered truth table. It always matched zero elements, so every output silently collapsed to a hardcoded constant 0.
- `solveBooleanFunction()` wrote to `output.value`, a Vue ref that only exists inside the component,  not inside this plain JS module. Entering a Boolean expression directly threw a ReferenceError immediately.
- The interactive-table flag was never set. The old code signaled "use the editable truth table" by passing null as an output parameter; the Vue port dropped that parameter, so the null landed in the wrong argument slot and the check never fired.

### Screenshots of the UI changes (If any) -

[Screencast from 2026-09-16 21-57-35.webm](https://github.com/user-attachments/assets/65eb918b-e15e-4b6c-ad49-1736b9f2ab7c)

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

The implementation fixes Combinational Analysis by using Vue's reactive tableBody data instead of reading values from the DOM.

I updated solveBooleanFunction() to return its result, fixed the required arguments and imports, and added outputStartCol to correctly identify output columns. Output cells in the truth table are now editable and cycle through 0 → 1 → x.

I chose the reactive-data approach because it is simpler and more reliable than DOM scraping. Tests were also updated to verify circuit generation and the corrected AND-gate result.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

- **New Features**
  - Truth tables can now be edited directly in the combinational analysis dialog.
  - Click output cells to cycle between `0`, `1`, and `x` values.
  - Circuits can be generated from edited truth tables, including “don’t care” entries.
  - Boolean-function analysis now displays its calculated output.

- **Bug Fixes**
  - Improved handling of truth-table outputs and validation messages.
  - Replaced inline alerts with consistent confirmation dialogs for invalid or unsupported inputs.

### CODE REVIEW

File: src/components/DialogBox/BooleanTable.vue [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
                 </th>
             </tr>
-            <tr v-for="tableRow in tableBody" :key="tableRow">
-                <th v-for="tableElement in tableRow" :key="tableElement">
+            <tr v-for="(tableRow, rowIndex) in tableBody" :key="rowIndex">
+                <th
+                    v-for="(tableElement, colIndex) in tableRow"
+                    :key="colIndex"
+                    :class="{ 'boolean-output-cell': isEditableCell(colIndex) }"
+                    @click="isEditableCell(colIndex) && cycleValue(rowIndex, colIndex)"
```
[the remark this answers is not in the document]
Comment (ShinichiShi) [reply src/components/DialogBox/BooleanTable.vue:16]: addressed

**Comment by ShinichiShi** [comment]
@coderabbitai review

**Review by tachyons — APPROVED** [review]
